src/main/java/com/googleinterns/zoomtube/servlets/TranscriptServlet.java

```diff
@@ -37,42 +52,123 @@
+      throws IOException {
+    String videoId = request.getParameter(PARAM_VIDEO_ID);
+    String transcriptXMLUrl = TRANSCRIPT_XML_URL_TEMPLATE + videoId;
+
     final Document document;
     try {
-      documentBuilder = documentBuilderFactory.newDocumentBuilder();
+      DocumentBuilder documentBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
       document = documentBuilder.parse(new URL(transcriptXMLUrl).openStream());
       document.getDocumentElement().normalize();
     } catch (ParserConfigurationException | SAXException e) {
       // TODO: Alert the user.
       System.out.println("XML parsing error");
-      return;
+      return Optional.empty();
     }
-    NodeList nodeList = document.getElementsByTagName(TEXT_TAG);
+    return Optional.of(document);
+  }
+
+  /**
+   * Puts each transcript line from {@code document} in datastore as its own entity. The lecture
+   * key to group the transcript lines under is indicated in {@code request}.
+   */
+  private void putTranscriptLinesInDatastore(HttpServletRequest request, Document document) {
+    long lectureId = Long.parseLong(request.getParameter(PARAM_LECTURE_ID));
+    NodeList nodeList = document.getElementsByTagName(TAG_TEXT);
     for (int nodeIndex = 0; nodeIndex < nodeList.getLength(); nodeIndex++) {
       Node node = nodeList.item(nodeIndex);
-      Element element = (Element) node;
-      String lineStart = element.getAttribute(START_ATTRIBUTE);
-      String lineDuration = element.getAttribute(DURATION_ATTRIBUTE);
-      String lineContent = node.getTextContent();
-      // TODO: Remove print statement. It is currently here for display purposes.
-      System.out.println(lineStart + " " + lineDuration + " "
-          + " " + lineContent);
+      datastore.put(createLineEntity(node, lectureId));
     }
   }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    PreparedQuery preparedQuery = getLectureTranscriptQuery(request);
+    ImmutableList<TranscriptLine> transcriptLines = getTranscriptLines(preparedQuery);
+    writeResponseAsJson(response, transcriptLines);
+  }
+
+  /**
+   * Gets the query for the lecture transcripts based on lecture id indicated in {@code request}.
+   */
+  private PreparedQuery getLectureTranscriptQuery(HttpServletRequest request) {
+    long lectureId = Long.parseLong(request.getParameter(PARAM_LECTURE_ID));
+    Key lecture = KeyFactory.createKey(PARAM_LECTURE, lectureId);
+
+    Filter lectureFilter =
+        new FilterPredicate(TranscriptLine.PROP_LECTURE, FilterOperator.EQUAL, lecture);
+
+    Query query = new Query(TranscriptLine.ENTITY_KIND)
+                      .setFilter(lectureFilter)
+                      .addSort(TranscriptLine.PROP_START, SortDirection.ASCENDING);
+    return datastore.prepare(query);
+  }
+
+  /**
+   * Gets the TranscriptLines from the {@code preparedQuery}.
```

Comment:
```suggestion
   * Returns the TranscriptLines from the {@code preparedQuery}.
```

Reply:
This comment seem redundant given the name of the method indicates what is happened. Also note that private methods need not have JavaDocs.

Reply:
Understood! I removed it.

Done!